pkg/dispatcher/dispatcher.go

```diff
@@ -39,6 +40,14 @@ const (
 	EventTypeScheduler
 )
 
+var (
+	ErrExcessAsyncDispatchLimit = errors.New("dispatcher exceeds async-dispatch limit")
+	AsyncDispatchLimit int32 = 10000
```

Comment:
Can we use the simple `fmt.Errorf()` call in the panic directly. This does not make it any clearer.

I also think that a hard coded limit of 10,000 could be a bit low compared to the queue length of 1,000,000. Go routines are light weight and do not add a lot of overhead. We can change the channel size already. If the channel has been increased to 100,000 via the config does it make sense to have an async buffer of 10,000 beside it? I am not sure that it would really help, maybe we need to set it based on 10% of the queue length.

Reply:
Thanks for these suggestions, make sense to me.